Sources/ContainerClient/Core/Volume.swift

```diff
@@ -0,0 +1,223 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the container project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// A named volume that can be mounted in containers.
+public struct Volume: Sendable, Codable, Equatable {
+    public var name: String
+    // Driver used to create the volume.
+    public var driver: String
+    // The mount point of the volume on the host.
+    public var mountpoint: String
+    public var createdAt: Date
+    // User-defined key/value metadata.
+    public var labels: [String: String]
+    // Driver-specific options.
+    public var options: [String: String]
+    public var scope: VolumeScope
+    public var size: UInt64?
+    public var usageData: VolumeUsageData?
+
+    public init(
+        name: String,
+        driver: String = "local",
+        mountpoint: String,
+        createdAt: Date = Date(),
+        labels: [String: String] = [:],
+        options: [String: String] = [:],
+        scope: VolumeScope = .local,
+        size: UInt64? = nil,
+        usageData: VolumeUsageData? = nil
+    ) {
+        self.name = name
+        self.driver = driver
+        self.mountpoint = mountpoint
+        self.createdAt = createdAt
+        self.labels = labels
+        self.options = options
+        self.scope = scope
+        self.size = size
+        self.usageData = usageData
+    }
+}
+
+/// Scope of volume availability.
+public enum VolumeScope: String, Sendable, Codable, CaseIterable {
```

Comment:
As with the options, I'm wondering if we keep the model simple to begin with – dealing with just our initial named volume implementation and the limited set of mount opts I'd expect us to support – and just make sure that what we do is open for extension.